File: Vastin/Frontend/vastin-video-player/types/Video.ts
import { User } from "@/types/User";

export type Video = {
    id: number,
    title: string | undefined,
    description: string | undefined,
    length: number | 0,
    videoPath: string | undefined,
    thumbnailPath: string | undefined,
    visibility: Visibility,
    createdAt: string,
    owner: User
};

enum Visibility {
    Public = 0,
    Private = 1,
    Unlisted = 2
}

export function visibilityResolver(input: Visibility ): string {
    switch (input) {
        case Visibility.Public:
            return "Public";
        case Visibility.Private:
            return "Private";
        case Visibility.Unlisted:
            return "Unlisted";
        default:
            return "Unknown";
    }
} 

File: Vastin/Models/Video.cs
                                                                                                                                                                                                                                                                                                                                                        using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Vastin.Models;

public enum VideoVisibility
{
    Public,
    Private,
    Unlisted
}

public class Video
{
    [Key]
    public int Id { get; set; }
    
    [Required]
    [MaxLength(200)]
    public string Title { get; set; }
    
    [MaxLength(2000)]
    public string Description { get; set; }
    
    [Required]
    public int Length { get; set; }
    
    [Required]
    public string VideoPath { get; set; }
    
    public string ThumbnailPath { get; set; }
    
    [Required]
    public VideoVisibility Visibility { get; set; } = VideoVisibility.Public;
    
    [Required]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    [ForeignKey("User_Video")]
    public User Owner { get; set; }
    
    public int OwnerId { get; set; }
}